airbyte-cdk/python/unit_tests/sources/declarative/incremental/test_per_partition_cursor_integration.py

```diff
@@ -232,7 +232,7 @@ def test_substream_without_input_state():
 
     stream_instance = test_source.streams({})[1]
 
-    stream_slice = StreamSlice(partition={"parent_id": "1"},
+    stream_slice = StreamSlice(partition={},
```

Comment:
This seems like a breaking change. Why has this changed?

Reply:
@maxi297 ah yes sorry i should've added a PR comment ahead of time for this. It's a bit of a tale.

So we mocked output for the parent stream's `read_records()` method (it overwrites `_read_pages()` but basically equates to the response used by the `SubstreamPartitionRouter`'s call to `parent_stream.read_records()`.

The problem here is that this `stream_slice` here is actually defined incorrectly. I dug into the code and in the dependent incremental parent stream, when we call `stream_slices()` which returns `StreamSlice`s where there is a time window for `cursor_value` and no `partition`. Which is expected since its just a plain incremental stream.

This only went uncaught because in the old implementation of `SubstreamPartitionRouter.stream_slices()`, we assign the `parent_slice` of the resulting StreamSlices by calling `parent_stream.stream_slices()` and extracting `parent_stream_slice.partition`. This correctly returned:
```
StreamSlice(partition={}, cursor_slice={"start_time": "2022-01-01", "end_time": "2022-01-31"})
```
and this would get added to the resulting final slices in `parent_slice`.

However, with the change where we no longer get parent records by calling `parent_stream.stream_slices()` + `parent_stream.read_records()` in favor of the higher level `read_stateless()`, we now have to inspect the record's associated_slice to populate `parent_slice`.
```
Record({"id": "1", CURSOR_FIELD: "2022-01-15"}, StreamSlice(partition={"parent_id": 1},
                               cursor_slice={"start_time": "2022-01-01", "end_time": "2022-01-31"}))
```

TLDR: we were incorrectly populating the `partition` field in the mocked records which caused the test to fail. So I fixed the incorrect mocked input which was only apparent w/ this new change

Reply:
What a journey! Thanks to your detailed comment, I get it now. I think the test setup confused me a bit even though it is appropriate. Can we add a comment over the `patch.object` to make it explicit that it will mock the parent stream `Rates` HTTP responses? Or rename `stream_slice` to `parent_stream_slice`? If you see any wait to make the test more explicit as well, feel free to chime in!

Reply:
yep i'll add a comment. it'll definitely be helpful given even i had to refresh myself yesterday for why i did this a week ago